shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -51,13 +52,14 @@ private static void ProcessType(StringBuilder builder, Type type, bool fullName)
             {
                 ProcessArrayType(builder, type, fullName);
             }
-            else if (_builtInTypeNames.TryGetValue(type, out var buildInName))
+            else if (_builtInTypeNames.TryGetValue(type, out var builtInName))
             {
-                builder.Append(buildInName);
+                builder.Append(builtInName);
             }
             else
             {
-                builder.Append(fullName ? type.FullName : type.Name);
+                var name = fullName || type.IsGenericParameter ? type.FullName : type.Name;
```

Comment:
If `type.IsGenericParameter` then skip fully.
for typeof(List<>), we want to print `List<>` for full name and short name both. With this code shortname will print `List<T>`

Reply:
For short name this code also will print `List<>`, there is a unit test for it [here](https://github.com/aspnet/Common/pull/283/files#diff-65ffadf8398a22a547ef1fc2cf411cb5R135), but I see your point, skipping fully would be easier.